src/raft.c

```diff
@@ -518,6 +530,31 @@ static void unlockDeleteKeys(RedisRaftCtx *rr, raft_entry_t *entry)
     }
 }
 
+static void freeSession(void *session)
+{
+    RedisModule_Free(session);
+}
+
+static void endSession(RedisRaftCtx *rr, raft_entry_t *entry)
+{
+    RedisModule_Assert(entry->type == RAFT_LOGTYPE_END_SESSION);
+    RaftReq *req = entry->user_data;
+
+    unsigned long long id = entry->session;
+
+    void *session = NULL;
+    RedisModule_DictDelC(rr->session_dict, &id, sizeof(id), &session);
```

Comment:
Agree that the implementation of `RedisModule_DictDelC` do it, but we should stick to the spec, and verify also the return value

Reply:
I believe this is fine?  session = NULL. after DictDelC, either it will still be NULL (no session set for this id) or it won't be NULL (yes, session).

as we then do

    if (session) {
        freeSession(session);
    }
    
it should all be fine.

Reply:
@fadidahanna ?